docs/coding-guidelines/add-types-to-SerDe.md

```diff
@@ -0,0 +1,100 @@
+# Add New Types to Serialization and Deserializaiton
+
+This how-to provides general instructions on how to add new types to serialization and deserialization between CLR (Common Language Runtime) and JVM (Java virtual machine). The IPC (Inter-process communication) mechanism between the CLR and the JVM uses serialization to communicate. Each type has a type identifier and then an optional length followed by the actual data.
+
+Not every type has been implemented so if you need to use a type that isn't supported and you get the error "Type {0} not supported yet", you will need to implement the type.
+
+## Steps to follow
+
+### 1. Choose a new type identifier and add it to:
+
+https://github.com/dotnet/spark/blob/master/src/csharp/Microsoft.Spark/Interop/Ipc/PayloadHelper.cs#L20
+
+Example:
+
+```csharp
+
+    private static readonly byte[] s_doubleArrayArrayTypeId = new[] { ( byte)'A' };z
+
+```
```

Comment:
Can you remove the empty lines and adjust the indentation? Also please remove `z` in the end.